cql3/statements/index_target.cc

```diff
@@ -71,6 +73,22 @@ index_target::target_type index_target::from_sstring(const sstring& s)
     throw std::runtime_error(format("Unknown target type: {}", s));
 }
 
+index_target::target_type index_target::from_target_string(const sstring& target) {
+    std::cmatch match;
+    if (std::regex_match(target.data(), match, target_regex)) {
+        return index_target::from_sstring(match[1].str());
+    }
+    return target_type::regular_values;
```

Comment:
If the string doesn't fit in the regex, then it is a name of the target column. So the only way to distinguish valid target from garbage (where would this garbage come from?) would be to check if a string is a valid name for column. Do we want to do it?

Reply:
Hmm, what happens if a column's name fits the regex? As far as I remember (am I misremembering something?), the name "values(x)" is a valid column name - it just needs to be quoted :-(
Can you please create a test with a regular column called "values(x)" and then try to index that column?
I'm obviously not suggesting that this is an important use case, but if we have such a bug, we should know about ir (and perhaps fix it now because we dig a backward-compatibility hole for ourselves).

Reply:
```
cqlsh:ks> create table "values(t)"(pk int, ck int, primary key(pk));
InvalidRequest: Error from server: code=2200 [Invalid query] message=""values(t)" is not a valid table name (must be alphanumeric character only: [0-9A-Za-z]+)"
```
am I doing something wrong?

Reply:
CQL's rules for table names and column names are different: Table names indeed must be alphanumeric ([9-9A-Za-z[+), but for column names, there is no such restriction - it just needs to be quoted if it contains weird characters. 
See for example https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/ref-lexical-valid-chars.html